### DISCUSSION TITLE
Subscriptions with Hasura as microservices

### DISCUSSION THREAD

**Discussion by cbarlow1993** [hasura/graphql-engine#9515] [Question]
Hi,

I'd like to know how it might be possible to support subscriptions and websocket connections following a microservices model described [https://hasura.io/docs/latest/deployment/performance-tuning/#microservices](https://hasura.io/docs/latest/deployment/performance-tuning/#microservices). 

Furthermore, is there any way to route websocket subscriptions through any other websocket service, maybe AWS Websocket API Gateway. Is there any metadata / records of which clients/headers have subscribed to websocket data.

This is primarily for Hasura Enterprise self hosted. 

Thanks
Chris

**Comment by meetzaveri** [comment]
Hey @cbarlow1993 

Thanks for asking. Before I proceed further, I would like to know what's the use-case for routing websocket subscriptions through any other websocket service (eg. AWS WS API gateway). Would love your inputs.

**Reply by cbarlow1993** [reply]
Hi @meetzaveri, predominantly it's due to our planned infrastructure. We currently use API gateway and a custom authorizer to validate our requests. This is perfect for standard HTTP requests and protects all our "compute" resources downstream at our network perimeter. Unfortunately, API Gateways are stateless and cannot persist WebSocket connections. 

We would love to use the AWS Websocket API Gateway infrastructure and it's serverless implementation if possible, but can't identify a specific way of doing so. 

Furthermore, support for remote schemas so that we could do websockets with chained hasura instances or sub graphs would be very powerful. It would allow our developers to separate our hasura deployments into smaller code bases and not impact each development team when making changes.

**Reply by meetzaveri** [reply]
Hey @cbarlow1993 . Thanks for your input. Unfortunately, there is a limitation for remote schemas, as you cannot perform subscriptions against remote schema fields. Just to let you know.

Let me discuss about AWS Websocket API gateway implementation with internal team members to gather more information.

**Reply by cbarlow1993** [reply]
Hi @meetzaveri , was there any solutions identified for this? Many thanks.
